community/kernel/src/main/java/org/neo4j/kernel/impl/util/ArrayQueueOutOfOrderSequence.java

```diff
@@ -96,6 +99,24 @@ public long highestEverSeen()
         return createResult( number, meta );
     }
 
+    @Override
+    public synchronized void await( long awaitedNumber, long timeoutMillis ) throws TimeoutException, InterruptedException
+    {
+        long endTime = System.currentTimeMillis() + timeoutMillis;
```

Comment:
I think value of `timeoutMillis` should be verified somewhere to avoid overflow of `endTime`